sklearn/utils/extmath.py

```diff
@@ -688,21 +688,41 @@ def _incremental_mean_and_var(X, last_mean=.0, last_variance=None,
     # old = stats until now
     # new = the current increment
     # updated = the aggregated stats
+    if not isinstance(last_sample_count, np.ndarray):
```

Comment:
@glemaitre do you mean that I change `last_sample_count` and `update_sample_count` to a ndarray? this will require to make changes in various files like:
* `sklearn/decomposition/incremental_pca.py`
* `sklearn/decomposition/tests/test_incremental_pca.py`
* `sklearn/preprocessing/tests/test_data.py`
* `sklearn/utils/tests/test_extmath.py`